src/script.js

```diff
@@ -0,0 +1,388 @@
+        category,
+        value,
+      }
+
+      filterCharacters(currentQuestion); 
+      question.selectedIndex = 0;
+    })
+  })
+}
+
+// It will filter the characters array and redraw the game board.
+const filterCharacters = (currentQuestion) => {
+  const secretCharacterValues = secretCharacter.hair === currentQuestion.value || secretCharacter.eyes === currentQuestion.value || secretCharacter.accessories.includes(currentQuestion.value) || secretCharacter.other.includes(currentQuestion.value);
+  const isArray = currentQuestion.category === 'accessories' || currentQuestion.category === 'other';
+
+  if (isArray) {
+    charactersInPlay = charactersInPlay.filter((character) => character[currentQuestion.category].includes(currentQuestion.value) === secretCharacterValues);
+  } else {
+    charactersInPlay = charactersInPlay.filter((character) => (character[currentQuestion.category] === currentQuestion.value) === secretCharacterValues);
+  }
+
+  alertMessage(secretCharacterValues, currentQuestion.value, currentQuestion.category);
+  generateBoard();
+  guessCount++
+  totalGuesses.innerHTML = `Number of guesses: ${guessCount}`
+  messageWrapper.style.display = "block";
+}
+
+const alertMessage = (keepCharacter) => {
+  const { category, value } = currentQuestion;
+  
+  if (category === 'accessories') {
+    if (keepCharacter) {
+      alertMessageText.innerHTML = `Yes, the person wears ${value}! Keep all people that wear ${value}.`;
+    } else {
+      alertMessageText.innerHTML = `No, the person doesn't wear ${value}! Remove all people that wear ${value}.`;
+    }
+  } else if (category === 'other') {
+    if (keepCharacter) {
+      alertMessageText.innerHTML = `Yes, the person is a ${value}! Keep all people that are ${value}s.`;
+    } else {
+      alertMessageText.innerHTML = `No, the person is not a ${value}! Remove all people that are ${value}s.`;
+    }
+  } else {
+    if (keepCharacter) {
+      alertMessageText.innerHTML = 
+        `Yes, the person has ${value} ${category}! Keep all people with ${value} ${category}.`;
+    } else {
+      alertMessageText.innerHTML = 
+        `No, the person doesnt have ${value} ${category}! Remove all people with ${value} ${category}.`;
+    }
+  }
+  generateBoard()
+}
+
+const guess = (personToConfirm) => {
+  checkMyGuess(personToConfirm)
+}
+
+// If you confirm, this function is invoked
+const checkMyGuess = (personToCheck) => {
+  if (secretCharacter.name === personToCheck) {
+    winOrLoseWrapper.style.display = "flex";
+    winOrLoseText.innerHTML = `<span class="emoji"> 🥳 </span> Great Job! It was ${secretCharacter.name}.`
+    gameBoard.innerHTML = ``
+    questionSection.innerHTML = ""
+  } else {
+    winOrLoseWrapper.style.display = "flex";
+    winOrLoseText.innerHTML = `<span class="emoji"> 😓 </span> It is not ${personToCheck}. It was ${secretCharacter.name}.`
+    gameBoard.innerHTML = ``
+    questionSection.innerHTML = ""
+  }
+}
+
+const startTimer = () => {
```

Comment:
curious question, is this added at the bottom in order for it to be read by the browser once the rest is in place?